index.bs

```diff
@@ -1432,8 +1429,131 @@ following additional fields:
+    sample rate.  They can differ by codec and related properties,
+    but should be different encodings of the same audio.
+
+: video
+:: A list of video encodings offered.  A video encoding is a series of
+    encoded video frames.  Encodings define properties needed by the
+    receiver to know how to decode the encoding, such as codec and
+    default duration.  They can differ by codec and potentially other
+    properties, but should be different encodings of the same video.
+
+
+Each audio encoding offered defines the following properties:
+
+: encoding-id
+:: Identifies the audio encoding being offered.  Must be unique within
+    the media stream.  Can be the receiver to request an encoding.
+
+: codec
+:: The name of the codec used by the encoding.
+
+: time-scale
+:: The time scale used by all audio frames.  This allows senders to
+    make audio-frame messages smaller by not including the time scale
+    in each one.
+
+: default-duration:
+:: The duration of an audio frame .  This allows senders to make
+    audio-frame messagse smaller by not including the duration for
+    audio-frame messages that have the default duration.
+
+Each video encoding offered defines the following properties:
+
+: encoding-id
+:: Identifies the video encoding being offered.  Must be unique within
+    the media stream.  Can be the receiver to request an encoding.
+
+: codec
+:: The name of the codec used by the encoding.
+
+: time-scale
+:: The time scale used by all video frames.  This allows senders to
+    make video-frame messages smaller by not including the time scale
+    in each one.
+
+: default-duration:
+:: The default duration of a video frame .  This allows senders to make
+    video-frame messagse smaller by not including the duration for
+    video-frame messages that have the default duration.
+
+: default-rotation:
+:: The default rotation of a video frame .  This allows senders to make
+    video-frame messagse smaller by not including the rotation for
+    video-frame messages that have the default rotation.
+
+After receiving a streaming-session-start-request message, a receiver
+should send back a streaming-session-start-response message with the
+following properties:
+
+: stats-interval
+:: Indicates the frequency the sender should send stats messages to
+    the receiver.
+
+: stream-requests
+:: Indicates which media streams the receiver would like to receiver
+    from the sender.
+
+Each stream request contains the following properties:
+
+: media-stream-id
+:: The ID of the stream reqeusted.
+
+: audio (optional)
+:: The requested audio encoding, by encoding ID
+
+: video (optional)
+:: The requested video encoding, by encoding ID.  It may
+    include a target resolution and maximum frame rate.  The sender
+    should not exceed the maximum frame rate and should attempt to
+    send at the target bitrate, possibly exceeding it by a small amount.
+
+
+During a streaming session, the receiver can modify the requests it
```

Comment:
I am assuming this allows the receiver to request a new (audio,video) encoding pair, that was already offered by the sender in 

What if the sender wants to change the encodings it offered?  Should it tear down the current session and start anew?

Reply:
Yes, that would require a new session, at least currently.  Do we have a use case where a sender wants to offer new encodings but wouldn't want to restart a session?

Reply:
During remoting, the encoding can change for a given video element.  For example, ads may be encoded differently from the main content.

For mirroring, we may want to switch codecs according to network conditions and content (e.g., lossy vs. lossless).

Reply:
Hmmm... in that case we need to go back to the drawing board a bit and go with a sender-controlled codec where the receiver gives is capabilities and then the sender chooses.

Reply:
Let's land this PR and do that in a separate PR.